### ISSUE TITLE
Problem with CustomXMLParts

### ISSUE THREAD

**Issue by vjsoria** [sdllc/Basic-Excel-R-Toolkit#16]
Hi,

I am presenting a problem by inserting a CustomXMLPart, is successful only the first call to Add, but in the second the answer is "an error occurred processing the command. Excel is busy?" then excel(version 2013 64 bits) crashes.

```
 xmlString <-
        paste("<?xml version=\"1.0\" encoding=\"utf-8\" ?>" ,
         "<employees xmlns=\"http://schemas.microsoft.com/vsto/samples2\">" ,
             "<employee>" ,
                 "<name>Karina Leal</name>" ,
                 "<hireDate>1999-04-01</hireDate>" ,
                 "<title>Manager</title>" ,
             "</employee>" ,
         "</employees>", sep="")
EXCEL$Application$get_ActiveWorkbook()$get_CustomXMLParts()$Add(xmlString, missing)
<environment: 0x000000000adfd070>
attr(,"class")
[1] "_CustomXMLPart" "IDispatch"     
 xmlString =
         paste("<?xml version=\"1.0\" encoding=\"utf-8\" ?>" ,
         "<outputvars xmlns=\"http://schemas.microsoft.com/vsto/samples\">" ,
             "<outputvar>" ,
                 "<name>Karina Leal</name>" ,
                 "<hireDate>1999-04-01</hireDate>" ,
                 "<title>Manager</title>" ,
             "</outputvar>" ,
         "</outputvars>", sep="")
 EXCEL$Application$get_ActiveWorkbook()$get_CustomXMLParts()$Add(xmlString, missing)
(an error occurred processing the command.  is excel busy?)
```

Any idea what could be happening?

**Comment by duncanwerner** [comment]
That actually worked for me.  Is there something in your spreadsheet?  Try running this in a blank workbook and see if you have the same problem.

**Comment by duncanwerner** [comment]
Actually, it seems to be crashing on exit now.  Probably related.  I will see if I can figure it out.  I'm not sure if this is related, but you should probably not be passing "missing" to that function; just leave the parameter out 

```
EXCEL$Application$get_ActiveWorkbook()$get_CustomXMLParts()$Add(xmlString)
```

**Comment by vjsoria** [comment]
I've tried the same case in another computer with a clean installation of Excel and still gives me the same error.

1. In the first call to Add I get the following response:

```
<Environment: 0x000000000adfd070>
attr ("class")
[1] "_CustomXMLPart" "IDispatch"

```
Which tells me it was successful.

2. The problem comes in the second, third, etc. calls, I get:
`(An error occurred processing the command. Excel is busy?)`

I think it was the same that causes crash when leaving

**Comment by duncanwerner** [comment]
There's a reference counting bug.  Not in the object itself, but in the type library check (generating function lists).  I will put together a release as soon as I have done some testing on the fix.

**Comment by duncanwerner** [comment]
Fixed in release v0.71

**Comment by vjsoria** [comment]
Thanks Duncan, now works fine
